velox/connectors/hive/iceberg/IcebergDataSink.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/iceberg/IcebergDataSink.h"
+
+#include "velox/common/base/Fs.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/exec/OperatorUtils.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+namespace {
+#define WRITER_NON_RECLAIMABLE_SECTION_GUARD(index)       \
+  memory::NonReclaimableSectionGuard nonReclaimableGuard( \
+      writerInfo_[(index)]->nonReclaimableSectionHolder.get())
+} // namespace
+
+IcebergDataSink::IcebergDataSink(
+    facebook::velox::RowTypePtr inputType,
+    std::shared_ptr<const IcebergInsertTableHandle> insertTableHandle,
+    const facebook::velox::connector::ConnectorQueryCtx* connectorQueryCtx,
+    facebook::velox::connector::CommitStrategy commitStrategy,
+    const std::shared_ptr<const HiveConfig>& hiveConfig)
+    : HiveDataSink(
+          std::move(inputType),
+          std::move(insertTableHandle),
+          connectorQueryCtx,
+          commitStrategy,
+          hiveConfig) {}
+
+void IcebergDataSink::appendData(facebook::velox::RowVectorPtr input) {
+  checkRunning();
+
+  // Write to unpartitioned table.
+  if (!isPartitioned()) {
+    const auto index = ensureWriter(HiveWriterId::unpartitionedId());
+    write(index, input);
+    return;
+  }
+
+  dataChannels_ = getDataChannels(partitionChannels_, inputType_->size());
+
+  // Compute partition and bucket numbers.
+  computePartitionAndBucketIds(input);
+
+  // Lazy load all the input columns.
+  for (column_index_t i = 0; i < input->childrenSize(); ++i) {
+    input->childAt(i)->loadedVector();
+  }
+
+  // All inputs belong to a single non-bucketed partition. The partition id
+  // must be zero.
+  if (!isBucketed() && partitionIdGenerator_->numPartitions() == 1) {
+    const auto index = ensureWriter(HiveWriterId{0});
+    write(index, input);
+    return;
+  }
+
+  splitInputRowsAndEnsureWriters(input);
+
+  for (auto index = 0; index < writers_.size(); ++index) {
+    const vector_size_t partitionSize = partitionSizes_[index];
+    if (partitionSize == 0) {
+      continue;
+    }
+
+    RowVectorPtr writerInput = partitionSize == input->size()
+        ? input
+        : exec::wrap(partitionSize, partitionRows_[index], input);
+
+    write(index, writerInput);
+  }
+}
+
+void IcebergDataSink::write(size_t index, RowVectorPtr input) {
```

Comment:
Earlier I had overridden it due to `auto dataInput = makeDataInput(dataChannels_, input);` present in HiveDataSink since we don't want to filter out the partition channels in the case of Iceberg when writing the data. 

But now that I have overridden the getDataChannels method already, I can remove this override here. I forgot to remove this, my bad.